x/programs/rust/wasmlanche_sdk/src/program.rs

```diff
@@ -56,24 +59,23 @@ impl From<i64> for Program {
     }
 }
 
-/// Marshals arguments into byte slice which can be unpacked by the host.
-fn marshal_args(args: &[Box<dyn Argument>]) -> Vec<u8> {
-    use std::mem::size_of;
-    // Size of meta data for each argument
-    const META_SIZE: usize = size_of::<i64>() + 1;
+#[macro_export]
+macro_rules! serialize_params {
```

Comment:
not sure how else to serialize an unspecified # of arguments with different types in rust. Can `macro_rules` be used here @richardpringle ?